### PR TITLE
[C#] Simplify concepts

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR attempts to simplify the concepts usage in the `config.json` a bit. It is rather basic and not very granular. That is on purpose though, as I'd like this to be as simple as possible. We can always introduce more granularity later if we feel that is required.

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
One thing we could consider: every exercise except the `basics` one has a prerequisite on the `basics` concept. I don't mind the explicitness here, but if people don't like that, we could argue that any exercise that has other prerequisites besides the `basics` one does not need to add the `basics` one. If we combine that with the rule that any non-basics exercise should have at least one prerequisite, we automatically get a transitive prerequisite on the `basics` concept.

**Comment by iHiD** [comment]
I think you spending a day getting confliglet checking concept paths and adding warnings (not errors for now) if things are missing would be good.

And also maybe it outputting a nice JSON file of all the tracks and their dependencies, which we can chart.

**Comment by ErikSchierboom** [comment]
> I think you spending a day getting confliglet checking concept paths and adding warnings (not errors for now) if things are missing would be good.

Yes, this would be very nice. My only question is: how to best implement it. Essentially, these are v3 config files, so the concepts paths should only be checked for v3 files. I can probably add that functionality to configlet, where it works on both the old _and_ the new files. Seems like the best strategy I think.

What do you think about the proposed simplification itself?

File: languages/csharp/config.json [hunk 1]
```diff
@@ -12,81 +12,56 @@
...
         "slug": "enums-advanced",
         "uuid": "ad240550-d732-4125-b00b-b852027a2db7",
-        "concepts": ["enums-advanced"],
-        "prerequisites": [
-          "enums-basic",
-          "attributes-basic",
-          "bitwise-operations",
-          "numbers-integers"
-        ]
+        "concepts": ["enums-flags", "bitwise-operations"],
```
Comment (iHiD) [inline languages/csharp/config.json:51]: I think this should be `enum-flags`

File: languages/csharp/config.json [hunk 2]
```diff
@@ -12,81 +12,56 @@
...
-        "concepts": ["properties-basic"],
+        "concepts": ["properties"],
         "prerequisites": [
-          "numbers-basic",
-          "numbers-floating-point",
-          "classes-basic",
-          "enums-basic",
-          "exceptions-basic",
-          "type-conversion-numbers"
+          "basics",
```
Comment (iHiD) [inline languages/csharp/config.json:59]: I'm happy for this to be removed.

File: languages/csharp/config.json (hunk 1 above)
Comment (ErikSchierboom) [reply languages/csharp/config.json]: It should actually be `flag-enums`, as that is how they are referred to. Thanks.

File: languages/csharp/config.json (hunk 2 above)
Comment (ErikSchierboom) [reply languages/csharp/config.json]: Done

**Comment by iHiD** [comment]
I think this is a huge improvement. Thanks :)

You might want to do PRs to other tracks too with the rationale when you've written it up?

**Pull request merged**